base-math/truncdddf.c

```diff
@@ -66,15 +66,25 @@ CONVERT_WRAPPER(
 	          return SIGNBIT(a) ? -INFINITY : INFINITY;
 	      }
 	  }
-	else if (exp < -BINPOWOF10_LIMIT)	/* Obvious underflow.  */
+	else if (exp <= -BINPOWOF10_DENORM_DBL)	/* Obvious underflow.  */
 	  {
 	    if (DFP_EXCEPTIONS_ENABLED)
 	      DFP_HANDLE_EXCEPTIONS (FE_UNDERFLOW|FE_INEXACT);
 
 	    switch (fegetround())
 	      {
 	        case FE_TONEAREST:
-	          return SIGNBIT(a) ? -__DBL_DENORM_MIN__ : __DBL_DENORM_MIN__;
+	          mant = llabs (a_norm);
+	          if (exp < -BINPOWOF10_DENORM_DBL ||
+	                    ((exp == -BINPOWOF10_DENORM_DBL) &&
+	                     (mant >= 24703282292062300)))
```

Comment:
It's necessary to replace this number with a macro or code that generates this value.
It helps to understand why this is being used.

This should have been indented this way:
```
if (exp < -BINPOWOF10_DENORM_DBL
    || ((exp == -BINPOWOF10_DENORM_DBL)
        && (mant >= 24703282292062300)))
```

Reply:
Ok. Will I will change it.